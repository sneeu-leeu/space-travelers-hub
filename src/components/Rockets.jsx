import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRockets } from '../redux/rockets/rockets';

const Rockets = () => {
  const rockets = useSelector((state) => state.rocketsReducer);
  const dispatch = useDispatch();
  useEffect(() => {
    if (!rockets.length > 0) {
      dispatch(fetchRockets());
    }
  }, []);

  return (
    <div>
      <p>rocket</p>
    </div>
  );
};

export default Rockets;
