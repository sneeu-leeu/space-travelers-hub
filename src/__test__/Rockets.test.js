import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Rockets from '../components/Rockets';
import store from '../redux/configureStore';

describe('check if all component are rendered', () => {
  it('renders correctly ', () => {
    const tree = renderer.create(<Provider store={store}><Rockets /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
