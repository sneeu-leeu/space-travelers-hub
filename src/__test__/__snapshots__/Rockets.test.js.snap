// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if all component are rendered renders correctly  1`] = `
<div>
  <ul />
</div>
`;
